pulpcore/pulpcore/app/models/repository.py

```diff
@@ -323,8 +325,7 @@ def save(self, *args, **kwargs):
         """
         Save the version while setting the number automatically.
         """
-        with suppress(self.DoesNotExist):
-            self.number = self.repository.versions.latest().number + 1
+        self.number = self.repository.last_version + 1
```

Comment:
It would be an improvement here to ensure that if ``save()`` was called in the future for some reason, it wouldn't overwrite the number.

Reply:
But that would be above and beyond the scope of this fix, so don't feel obligated. :)